agents_infra/agents_infra/agents/dns/dns.py

```diff
@@ -69,35 +76,47 @@ def is_dns_running(self):
 
     def is_service_healthy(self, timeout=2, payload=None, packet_size=0):
         process_status = super(DNSAgent, self).is_service_healthy()
-        port = self.is_port_open(
-            timeout=timeout, payload=payload, packet_size=packet_size
+        port = is_port_open(
+            port=self.config.get('port'),
+            ip=self.ip,
+            protocol=self.protocol,
+            logger=self.logger,
+            timeout=timeout,
+            payload=payload,
+            packet_size=packet_size
         )
         dns_status = self.is_dns_running()
 
         return process_status and port and dns_status
 
-    def maybe_restart_service(self):
-        inactive_services = []
-        if not self.is_dns_running():
-            inactive_services.append('named')
-
-        if not self.is_ssh_service_active():
-            inactive_services.append('ssh')
-
-        if inactive_services:
-            for service in inactive_services:
-                restart_service(service, logger=self.logger)
 
-            maybe_log_message(
-                'Finished attempts to restart services',
-                self.logger,
-                level=logging.INFO,
-            )
-            return False
+class DNSAgentNamed(DNSAgent):
+    """
+    DNSAgentNamed is a specialized subclass of DNSAgent designed to monitor
+    and manage a DNS server running with the 'named' process.
+    """
 
-        maybe_log_message(
-            'All services are heathy and running',
-            self.logger,
-            level=logging.INFO,
+    def __init__(
+        self,
+        protocol='udp',
+        command_queue_size=0,
+        config=None
+    ):
+        # Setting ='dns_named' if not provided
+        if config is None:
+            config = Config(name='dns_named', protocol=protocol)
+        elif isinstance(config, dict):
+            config.setdefault('name', 'dns_named')
+            config.setdefault('protocol', protocol)
+            config = Config.from_dict(config)
+        elif isinstance(config, Config):
+            if not config.name:
+                config.name = 'dns_named'
+            if not getattr(config, 'protocol', None):
+                config.protocol = protocol
```

Comment:
First of all, see above the same remark about forcing `name` and `protocol` values when user passes an instance of `dict` or `Config`. Second, this is kinda duplicated logic - we violate DRY. But to resolve this we'd have to parametrize server name at each level of inheritance. Maybe, this could be addressed later :)